### DISCUSSION TITLE
LubuntuにMOZC UT2をインストールしたい

### DISCUSSION THREAD

**Discussion by miracle777** [google/mozc#621] [Q&A]
今までUbuntu18.04や20.04でUT2を使ってきました。
UT2のお陰で快適に日本語入力できるので、Lubuntu22.04にもインストールしたいです。

http://linuxplayers.g1.xrea.com/mozc-ut.html
を読みました。
`cat mozcdic-ut-20220823/mozcdic-ut-20220823.txt >> mozc-master/src/data/dictionary_oss/dictionary00.txt`
までは、出来ました。

そのあと、Build Mozc as usual.と書いてあります。
でも私は、MOZCのビルドの方法を知らないのです。

MOZCのビルドがbazelを使うところまでは、分かりました。
Lubuntuに、bazelのインストールが出来ました。

`bazel --version
bazel 5.3.0
`
今までは、このサイトの方法で、インストールしてました。
https://sicklylife.jp/ubuntu/2004/mozc_ut2.html

こうした記事も、見つけました。
http://blog.livedoor.jp/isooki/archives/52135251.html

ビルドに必要なパッケージのインストールで、pythonが無いとエラーになってしまいます。
実際には、python3が、インストールされています。
`python3 -V
Python 3.10.4`

Lubuntuなので、Dockerを使ったインストールもできなのは？と思っています。

Ubuntuのバージョンが22.04に上がっていることもあり、今の時点で、一番新しい本当のMOZC UT2のインストールの方法を知りたいです。
よろしくお願いします。

私は、Fcitx-mozc　を使っています。

**Comment by hiroyuki-komatsu** [comment]
Hi miracle777,

The build process is described here:

* [mozc/build_mozc_in_docker.md at master · google/mozc](https://github.com/google/mozc/blob/master/docs/build_mozc_in_docker.md)

GitHub Actions workflow is also probably helpful:

* [mozc/linux.yaml at master · google/mozc](https://github.com/google/mozc/blob/master/.github/workflows/linux.yaml)


Sorry for saying that, but we only support the above Docker build for focusing on the core application features rather than the build configurations for multiple distributions.

Moving this issue to the discussions, we hope there are discussions among users of Lubuntu/UT2/Fcitx.

**Reply by miracle777** [reply]
Thank you very much.
I will practice to get used to how to build using Docker.

I'm on Ubuntu 20.04 right now, so I did the build using Docker as soon as possible.

At first, there was no version number when converted to version.
After building, the version appeared.
Is the build successful in this situation?

![スクリーンショット_20221022_193317](https://user-images.githubusercontent.com/13194391/197335940-66b6141f-87dd-473d-9596-d14796fa73db.png)

**Reply by shitamo** [reply]
別にDockerを使う必要はないのではないでしょうか？
https://github.com/google/mozc/blob/master/.github/workflows/linux.yaml をみるとubuntu-20.04でビルドしている手順がわかると思います。

ただビルドするだけなら参考のとおりでいいですけど、ファイルの配置のことを考えるとパッケージをつくるほうが楽ですよね。
とりあえず手抜きしまくりですが
https://github.com/shitamo/mozc/tree/debian
にパッケージをつくるためのものを入れてみました。
VirtualBox上のLubuntuでパッケージはできた感じです。
UT2の辞書はいれてませんけど…。パッケージが作れたら次の手順として入れられるんじゃないでしょうか。
バージョンの更新のたびに変更するのが面倒だったのでdebianブランチは消してたのでまたそのうち消してしまうと思います。

使い方としては

> git clone https://github.com/shitamo/mozc.git -b debian --single-branch --recursive
> cd mozc
> dpkg-buildpackage -us -uc -b

**Reply by miracle777** [reply]
うーん・・・
どうなのでしょう。
UT2のインストールの方法が、Dockerに変わった理由があるようなので。

最初は、こちら。
https://sicklylife.jp/ubuntu/2004/mozc_ut2.html

それから、このサイトを知りました。
http://linuxplayers.g1.xrea.com/mozc-ut.html

Dockerでなくてもよいのですが、UT２が容易に利用できるようになると、Ubuntuの利用者も増えると思います。

今回たまたまUbuntu20.04でしたので、Dockerでなくても良かったと思います。

Ubuntu18.04に先日UT2をインストールを試みたときに、従来の方法でインストールできませんでした。

本当は、MOZCの辞書が充実していたらUT2を追加でインストールしなくてもよいです。
WindowsやMacのように快適に日本語変換しようとするとUT2が、必要に感じています。

Ubuntuが普及するために、日本語環境でUT2をインストールしなくても良いようになってほしいと願います。

**Reply by shitamo** [reply]
Dockerというよりたんに環境が変わっただけじゃないですかね。たとえばpython3になったけど古いソースはpython2.7を想定してたとか。
Mozcの辞書については要望を出してみるとか…対応してくれるかは不明ですが。

**Comment by miracle777** [comment]
[hiroyuki-komatsu](https://github.com/hiroyuki-komatsu)さんが、ビルドの環境を統一したいので、Dockerを利用すると言っていました。
下記が原文です。
Sorry for saying that, but we only support the above Docker build for focusing on the core application features rather than the build configurations for multiple distributions.

ありがとうございます。
Mozcの辞書のことは、要望伝えることができる窓口探してみます。
